extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -0,0 +1,218 @@
+            // that mongoClient will use putObjectNoVer which just puts
+            // the object without further manipulation/actions.
+            // S3 takes care of the versioning logic so consuming the queue
+            // is sufficient to replay the version logic in the consumer.
+            return this._mongoClient.putObject(bucket, key, objVal, undefined,
+                this.logger, err => {
+                    if (err) {
+                        this.logger.error('error putting object to mongo',
+                        { error: err });
+                        return done(err);
+                    }
+                    this.logger.info('successfully put object to mongo',
+                    { key });
+                    return done();
+                });
+        }
+        if (sourceEntry instanceof BucketMdQueueEntry) {
+            const masterBucket = sourceEntry.getMasterBucket();
+            const instanceBucket = sourceEntry.getInstanceBucket();
+            const val = sourceEntry.getValue();
+            return this._mongoClient.putObject(masterBucket,
+                instanceBucket, val, undefined,
+                this.logger, err => {
+                    if (err) {
+                        this.logger.error('error putting bucket ' +
+                        'metadata to mongo',
+                        { error: err, masterBucket, instanceBucket });
+                        return done(err);
+                    }
+                    this.logger.info('successfully put bucket' +
+                    ' metadata to mongo', { masterBucket, instanceBucket });
+                    return done();
+                });
+        }
+        if (sourceEntry instanceof BucketQueueEntry) {
+            const bucketOwnerKey = sourceEntry.getBucketOwnerKey();
+            const val = sourceEntry.getValue();
+            return this._mongoClient.putObject(usersBucket,
+                bucketOwnerKey, val, undefined,
+                this.logger, err => {
+                    if (err) {
+                        this.logger.error('error putting to usersbucket',
+                        { error: err, bucketOwnerKey });
+                        return done(err);
+                    }
+                    this.logger.info('successfully put to usersBucket',
+                    { bucketOwnerKey });
+                    return done();
+                });
+        }
+        this.logger.warn('skipping unknown source entry',
+                            { entry: sourceEntry.getLogInfo() });
+        return process.nextTick(done);
+    }
+
+    /**
+     * Bootstrap kafka consumer by periodically sending bootstrap
+     * messages and wait until it's receiving newly produced messages
+     * in a timely fashion. ONLY USE FOR TESTING PURPOSE.
+     *
+     * @param {function} cb - callback when consumer is effectively
+     * receiving newly produced messages
+     * @return {undefined}
+     */
+    bootstrapKafkaConsumer(cb) {
```

Comment:
I think this function is no longer useful, instead the `bootstrap` boolean can be passed to `BackbeatConsumer` params for tests.